kernel/src/main/java/org/kframework/parser/concrete2kore/kernel/Scanner.java

```diff
@@ -52,11 +53,44 @@ public Module getModule() {
         return module;
     }
 
+    public Set<Integer> kinds() {
+        return tokens.values().stream().map(v -> v._1()).collect(Collectors.toSet());
+    }
+
     // debugging method
-    private TerminalLike getTokenByKind(int kind) {
+    public TerminalLike getTokenByKind(int kind) {
         return tokens.entrySet().stream().filter(e -> e.getValue()._1() == kind).findAny().get().getKey();
     }
 
+    public void getStandaloneScanner(File path) {
+        StringBuilder flex  = new StringBuilder();
+        flex.append("%{\n" +
+            "#include \"node.h\"\n" +
+            "#include \"parser.tab.h\"\n" +
+            "%}\n\n" +
+            "%option noyywrap\n" +
+            "%%\n\n");
+        if (this.module.allSorts().contains(Sorts.Layout())) {
+            flex.append(this.module.layout() + " ;\n");
+        }
+        List<TerminalLike> ordered = tokens.keySet().stream().sorted((t1, t2) -> tokens.get(t2)._2() - tokens.get(t1)._2()).collect(Collectors.toList());
+        for (TerminalLike key : ordered) {
+            if (key instanceof Terminal) {
+                Terminal t = (Terminal) key;
+                flex.append(StringUtil.enquoteCString(t.value()));
+            } else {
+                RegexTerminal t = (RegexTerminal) key;
+                flex.append(t.regex());
+            }
+            writeStandaloneAction(flex, key);
+        }
+        try {
+            FileUtils.write(path, flex);
+        } catch (IOException e) {
+            throw KEMException.internalError("Failed to write file for scanner", e);
+        }
+    }
```

Comment:
Is there no way for `getStandaloneScanner` and `getScanner` to re-use chunks amongst themselves?